_rules/SC2-4-6-descriptive-labels.md

```diff
@@ -0,0 +1,150 @@
+
+### Expectation
+
+Each target element describes the purpose of the associated form field element.
+
+**Note**: Labels do not need to be lengthy. A word, or even a single character, may suffice.
+
+## Assumptions
+
+_There are currently no assumptions._
+
+## Accessibility support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 2.4.6: Headings and Labels](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html) 
+- [G131: Providing descriptive labels](https://www.w3.org/WAI/WCAG21/Techniques/general/G131)
+- [H44: Using label elements to associate text labels with form controls](https://www.w3.org/WAI/WCAG21/Techniques/html/H44)
+- [ARIA16: Using aria-labelledby to provide a name for user interface controls](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA16)
+
+## Test Cases
+
+### Passed
+
+#### Passed example 1
+
+Label that is coded with the `label` element and describes the purpose of the associated element.
+
+```html
+<label for="fname">First name:</label>
+<input id="fname" type="text" name="fname"/>
+```
+
+#### Passed example 2
+
+Label that is coded with the `p` element and associated by the aria-labelledby attribute. The label describes the purpose of the associated element.
+
+```html
+<p id="label_fname">First name:</p>
+<input aria-labelledby="label_fname" type="text" name="fname"/>
+```
+
+#### Passed example 3
+
+Implicit label that is coded with the `label` element and describes the purpose of the associated element.
+
+```html
+<label>First name:<input id="fname" type="text" name="fname"/></label>
+```
+
+### Failed
+
+#### Failed example 1
+
+Label that is coded with the `label` element and does not describe the purpose of the associated element.
+
+```html
+<label for="fname">Menu</label>
+<input id="fname" type="text" name="fname"/>
+```
+
+#### Failed example 2
+
+Label that is coded with the `p` element and associated by the aria-labelledby attribute. The label does not describe the purpose of the associated element.
+
+```html
+<p id="label_fname">Menu</p>
+<input aria-labelledby="label_fname" type="text" name="fname"/>
+```
+
+#### Failed example 3
+
+Implicit label that is coded with the `label` element and does not describe the purpose of the associated element.
+
+```html
+<label>Menu<input id="fname" type="text" name="fname"/></label>
+```
+
+### Inapplicable
+
+#### Inapplicable example 1
+
+`Label` that is neither visible to users, nor included in the accessibility tree.
+
+```html
+<label for="fname" style:"display:none;">First name:</label>
+<input id="fname" type="text" name="fname"/>
+```
+
+#### Inapplicable example 2
+
+Hidden `p` element that associated by the aria-labelledby attribute.
```

Comment:
Editorial: "element that IS associated" or "element associated"

Reply:
Editorial: `p` element that is neither visible nor included in the accessibility tree